lib/esi/calls.rb

```diff
@@ -34,12 +34,8 @@ class Base
 
       attr_accessor :path, :params
 
-      def name
-        @name ||= self.class.name.remove('Esi::Calls::').underscore.to_sym
-      end
-
       def cache_key
-        @cache_key ||= ActiveSupport::Cache.expand_cache_key([name, params].flatten, :esi)
+        @cache_key ||= ActiveSupport::Cache.expand_cache_key([path, params].flatten, :esi)
```

Comment:
I still think we should `to_set.hash` the params and path here, a string hash is going to be more compatible for users using something like redis for a cache store.

Reply:
That and this is failing on Travis right now

Reply:
because this is called in a base class we may need to call `@path` here, also another justification for `to_set.hash` is the following:

```
client.some_call(id: 1, page: 1).cache_key
#=> "[some_call, {id: 1, page: 1}]"

# keys will not match even though the underlying request is the same.
client.some_call(page: 1, id: 1).cache_key
#=> "[some_call, {page: 1, id: 1}]"
```

Reply:
Completely agree and `redis` is a very good argument which I didn't think about.